### ISSUE TITLE
Issue with "<input type="file">": (short summary here please)

### ISSUE THREAD

**Issue by dunkinman** [mdn/content#9420]
MDN URL: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/file

#### What information was incorrect, unhelpful, or incomplete?
The example on the page doesn't work.

#### Specific section or headline?
The bottom of the page shows an error.

#### What did you expect to see?
I was playing with this code...and copied it verbatim...but it doesn't produce anything.

#### Did you test this? If so, how?
Yes...I copied line for line...no output.  HTML works...but clicking on button after attaching documents yields nothing.

<details>
<summary>MDN Content page report details</summary>

* Folder: `en-us/web/html/element/input/file`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/file
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/html/element/input/file/index.md
* Last commit: https://github.com/mdn/content/commit/bbaf3676373af6f6ce9817141d25fbf7ba88ef8d
* Document last modified: 2021-09-29T14:51:58.000Z

</details>

**Comment by digi-booster** [comment]
What was the error and can you share your code?

**Comment by dunkinman** [comment]
I literally copied all of the code from the page and tried to run it....

<form method="post" enctype="multipart/form-data">
  <div>
    <label for="image_uploads">Choose images to upload (PNG, JPG)</label>
    <input type="file" id="image_uploads" name="image_uploads" accept=".jpg, .jpeg, .png" multiple>
  </div>
  <div class="preview">
    <p>No files currently selected for upload</p>
  </div>
  <div>
    <button>Submit</button>
  </div>
</form>

const input = document.querySelector('input');
const preview = document.querySelector('.preview');

input.style.opacity = 0;

input.addEventListener('change', updateImageDisplay);

function updateImageDisplay() {
  while(preview.firstChild) {
    preview.removeChild(preview.firstChild);
  }

  const curFiles = input.files;
  if(curFiles.length === 0) {
    const para = document.createElement('p');
    para.textContent = 'No files currently selected for upload';
    preview.appendChild(para);
  } else {
    const list = document.createElement('ol');
    preview.appendChild(list);

    for(const file of curFiles) {
      const listItem = document.createElement('li');
      const para = document.createElement('p');
      if(validFileType(file)) {
        para.textContent = `File name ${file.name}, file size ${returnFileSize(file.size)}.`;
        const image = document.createElement('img');
        image.src = URL.createObjectURL(file);

        listItem.appendChild(image);
        listItem.appendChild(para);
      } else {
        para.textContent = `File name ${file.name}: Not a valid file type. Update your selection.`;
        listItem.appendChild(para);
      }

      list.appendChild(listItem);
    }
  }
}

// https://developer.mozilla.org/en-US/docs/Web/Media/Formats/Image_types
const fileTypes = [
  "image/apng",
  "image/bmp",
  "image/gif",
  "image/jpeg",
  "image/pjpeg",
  "image/png",
  "image/svg+xml",
  "image/tiff",
  "image/webp",
  "image/x-icon"
];

function validFileType(file) {
  return fileTypes.includes(file.type);
}

I just checked the page and the demo works now...it wasn't working before.  Did you change something?

**Comment by digi-booster** [comment]
Add this at the end of your JavaScript code.
```js
function returnFileSize(number) {
  if(number < 1024) {
    return number + 'bytes';
  } else if(number >= 1024 && number < 1048576) {
    return (number/1024).toFixed(1) + 'KB';
  } else if(number >= 1048576) {
    return (number/1048576).toFixed(1) + 'MB';
  }
}
```

**Comment by dunkinman** [comment]
I forgot to copy that in here..but I did use it in my test.  The example at the bottom this am...was not working but now it is...was something changed?

**Comment by digi-booster** [comment]
> I forgot to copy that in here..but I did use it in my test. The example at the bottom this am...was not working but now it is...was something changed?

No, there's probably a glitch last time you tried it.

**Comment by dunkinman** [comment]
No.  Here's what's misleading...when you go to the bottom and try it it obviously works and shows the file sizes.  When you copy the code and try it it does not.  When you click on the see it running live it only allows for one file and it does not show anything either.

**Comment by dunkinman** [comment]
I actually do see that I clicked an example higher up on the page...and was clicking the wrong link...but when I stepped through...it is working on the page...but not when I'm copying in the code.  It's not showing me the file sizes.

**Comment by dunkinman** [comment]
Mystery solved.  I copied the code from the page...and it didn't work.  Once I went to Github and got it from there all is well.  Sorry for the trouble.  I'm a newb...but I also think it's a bit misleading that the page shows code...but that you have to get the code in it's entirety to be able to work with it.  Just food for thought from a newb.

**Comment by digi-booster** [comment]
I am not able to reproduce the error can you post your code online with the help of the [GitHub page](https://pages.github.com/), [Codepen](https://codepen.io/) or [jsfiddle](https://jsfiddle.net/)?
